mlflow/store/artifact/databricks_models_artifact_repo.py

```diff
@@ -156,13 +157,10 @@ def _parallelized_download_from_cloud(
                 env=parallel_download_subproc_env,
                 headers=headers,
             )
-            download_errors = [
-                e for e in failed_downloads.values() if e["error_status_code"] not in (401, 403)
-            ]
-            if download_errors:
+            if any(not e.retriable for e in failed_downloads.values()):
                 raise MlflowException(
                     f"Failed to download artifact {dst_run_relative_artifact_path}: "
-                    f"{download_errors}"
+                    f"{failed_downloads}"
```

Comment:
Now the exception doesn't contain the status_code, do you think it's useful in the error message? Maybe keep status_code but define .retriable() as a method?

Reply:
I think including the status_code can be misleading because we catch non-HTTP errors.

Reply:
Fair enough :D

Reply:
The status code is not included, but the HTTP error name is. We can infer the status code from the name.

Reply:
Added status code as an optional argument